src/coreclr/jit/codegenarmarch.cpp

```diff
@@ -2948,6 +2959,17 @@ void CodeGen::genCodeForCpBlkUnroll(GenTreeBlk* node)
     }
 #endif
 
+#ifndef JIT32_GCENCODER
+    if (!node->gtBlkOpGcUnsafe && ((srcOffsetAdjustment != 0) || (dstOffsetAdjustment != 0)))
+    {
+        // If node is not already marked as non-interruptible, and if are about to generate code
+        // that produce GC references in temporary registers not reported, then mark the block
+        // as non-interruptible.
+        node->gtBlkOpGcUnsafe = true;
+        GetEmitter()->emitDisableGC();
```

Comment:
Add a note that the corresponding `emitEnableGC()` call will happen in the caller?